projects/angular/src/forms/number-input/number-input-container.ts

```diff
@@ -0,0 +1,92 @@
+import { LayoutService } from '../common/providers/layout.service';
+import { NgControlService } from '../common/providers/ng-control.service';
+import { ClrNumberInput } from './number-input';
+
+@Component({
+  selector: 'clr-number-input-container',
+  template: `
+    <ng-content select="label"></ng-content>
+    <label *ngIf="!label && addGrid()"></label>
+    <div class="clr-control-container" [ngClass]="controlClass()">
+      <div class="clr-number-input-wrapper">
+        <div class="clr-input-group" [class.clr-focus]="focus">
+          <ng-content select="[clrNumberInput]"></ng-content>
+          <div class="clr-input-group-actions">
+            <button
+              type="button"
+              class="clr-input-group-icon-action"
+              (click)="input.stepDown($event)"
+              [disabled]="control?.disabled"
+            >
+              <cds-icon shape="minus" size="sm"></cds-icon>
+            </button>
+            <div class="clr-number-input-separator"></div>
+            <button
+              type="button"
+              class="clr-input-group-icon-action"
+              (click)="input.stepUp($event)"
+              [disabled]="control?.disabled"
+            >
+              <cds-icon shape="plus" size="sm"></cds-icon>
+            </button>
+          </div>
+        </div>
+        <cds-icon
+          *ngIf="showInvalid"
+          class="clr-validate-icon"
+          shape="exclamation-circle"
+          status="danger"
+          aria-hidden="true"
+        ></cds-icon>
+        <cds-icon
+          *ngIf="showValid"
+          class="clr-validate-icon"
+          shape="check-circle"
+          status="success"
+          aria-hidden="true"
+        ></cds-icon>
+      </div>
+      <ng-content select="clr-control-helper" *ngIf="showHelper"></ng-content>
+      <ng-content select="clr-control-error" *ngIf="showInvalid"></ng-content>
+      <ng-content select="clr-control-success" *ngIf="showValid"></ng-content>
+    </div>
+  `,
+  host: {
+    '[class.clr-form-control]': 'true',
+    '[class.clr-form-control-disabled]': 'control?.disabled',
+    '[class.clr-form-control-readonly]': 'input.readonly',
+    '[class.clr-row]': 'addGrid()',
+  },
+  providers: [FocusService, IfControlStateService, NgControlService, ControlIdService, ControlClassService],
+})
+export class ClrNumberInputContainer extends ClrAbstractContainer {
+  focus = false;
+
+  @ContentChild(forwardRef(() => ClrNumberInput)) protected readonly input: ClrNumberInput;
+
+  constructor(
+    controlClassService: ControlClassService,
+    @Optional() layoutService: LayoutService,
+    ngControlService: NgControlService,
+    focusService: FocusService,
+    protected override ifControlStateService: IfControlStateService
+  ) {
+    super(ifControlStateService, layoutService, controlClassService, ngControlService);
+
+    this.subscriptions.push(focusService.focusChange.subscribe(state => (this.focus = state)));
```

Comment:
This should be in `ngOnInit`, not the constructor.

Reply:
This matches existing coding patterns. We should change it everywhere in a separate PR.